### PR TITLE
Blog app - integration specs for views and fix n+1 problems

### PR DESCRIPTION
**Pull request by fibini**
Make sure that the N+1 problem is solved when fetching all posts and their comments for a user. 
Integration specs
- Use Capybara to write integration tests for each view in your project.

User index page:

- I can see the username of all other users.
- I can see the profile picture for each user.
- I can see the number of posts each user has written.
- When I click on a user, I am redirected to that user's show page.

User show page:
- I can see the user's profile picture.
- I can see the user's username.
- I can see the number of posts the user has written.
- I can see the user's bio.
- I can see the user's first 3 posts.
- I can see a button that lets me view all of a user's posts.
- When I click a user's post, it redirects me to that post's show page.
- When I click to see all posts, it redirects me to the user's post's index page.

User post index page:
- I can see the user's profile picture.
- I can see the user's username.
- I can see the number of posts the user has written.
- I can see a post's title.
- I can see some of the post's body.
- I can see the first comments on a post.
- I can see how many comments a post has.
- I can see how many likes a post has.
- I can see a section for pagination if there are more posts than fit on the view.
- When I click on a post, it redirects me to that post's show page.

Post show page:
- I can see the post's title.
-  can see who wrote the post.
- I can see how many comments it has.
- I can see how many likes it has.
- I can see the post body.
- I can see the username of each commentor.
- I can see the comment each commentor left.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -37,6 +37,13 @@ Cloning a repository
...
+👤 **Dino Quispe**
+
+- GitHub: [@dqarias](https://github.com/dqarias)
+- Twitter: [@DinoRonald7](https://twitter.com/DinoRonald7?t=Zanx9DXMEG9C_PNF3woZFg&s=08)
+- LinkedIn: [Dino Quispe](https://www.linkedin.com/in/dino-ronald-quispe-arias/)
+
+
 ## Show your support
 
 Give a ⭐️ if you like this project!
```
Comment (omar-muhamad) [inline README.md:57]: - [ ] Please add the instructions about how to `install required gems`, `create and run migrations for the database`, and `how to run tests` under Getting started section in your readme file to make it more professional.

File: spec/views/users/index.html.erb_spec.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    end
+    it 'show the user name and show the user photo' do
+      visit users_path
+      sleep(6)
+      expect(page).to have_content(@first_user.name)
+      expect(page).to have_content(@second_user.name)
+      expect(page).to have_content(@third_user.name)
+      expect(page).to have_xpath("//img[contains(@src,'#{@first_user.photo}')]")
+      expect(page).to have_xpath("//img[contains(@src,'#{@second_user.photo}')]")
+      expect(page).to have_xpath("//img[contains(@src,'#{@third_user.photo}')]")
```
Comment (omar-muhamad) [inline spec/views/users/index.html.erb_spec.rb:26]: - [x] Please make these in a separate test case like the following:
```ruby
it 'show the user photo' do
    'Put the tests here'
end
```

File: spec/views/users/show.html.erb_spec.rb [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+      @third_post = Post.create(title: 'My third post', text: 'It is raining today', comments_counter: 0,
+                                likes_counter: 0, author: @first_user)
+    end
+    it 'shows the user photo, name, post count and bio' do
+      visit user_path(@first_user)
+      sleep(3)
+      expect(page).to have_content(@first_user.name)
+      expect(page).to have_content('number of posts: 3')
+      expect(page).to have_content(@first_user.bio)
+      expect(page).to have_xpath("//img[contains(@src,'#{@first_user.photo}')]")
```
Comment (omar-muhamad) [inline spec/views/users/show.html.erb_spec.rb]: - [x] Same here, Please put every one of these tests in a separate test case.

File: spec/views/users/show.html.erb_spec.rb [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+      sleep(3)
+      expect(@first_user.recent_post.length).to eql 3
+    end
+
+    it 'shows the see all post button and redirects to the post index page' do
+      visit user_path(@first_user)
+      sleep(3)
+      expect(page).to have_link('See all posts')
+      click_link 'See all posts'
+      expect(page).to have_current_path(user_posts_path(@first_user))
```
Comment (omar-muhamad) [inline spec/views/users/show.html.erb_spec.rb]: - [x]  Same here, Please put every one of these tests in a separate test case.

File: spec/views/posts/index.html.erb_spec.rb [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    end
+
+    it 'shows the user photo, name and post count' do
+      visit user_posts_path(@first_user, @first_post)
+      sleep(3)
+      expect(page).to have_content(@first_user.name)
+      expect(page).to have_content('number of posts: 1')
+      expect(page).to have_content(@first_post.title)
+      expect(page).to have_content(@first_post.text)
+      expect(page).to have_xpath("//img[contains(@src,'#{@first_user.photo}')]")
```
Comment (omar-muhamad) [inline spec/views/posts/index.html.erb_spec.rb]: - [x] Same here, Please put every one of these tests in a separate test case.

File: spec/views/posts/index.html.erb_spec.rb [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    end
+
+    it 'shows the post comments, likes count and comments count' do
+      visit user_posts_path(@first_user, @first_post)
+      sleep(3)
+      expect(page).to have_content(@first_comment.text)
+      expect(page).to have_content('Comments: 1')
+      expect(page).to have_content('Likes: 1')
```
Comment (omar-muhamad) [inline spec/views/posts/index.html.erb_spec.rb]: - [x]  Same here, Please put every one of these tests in a separate test case.

File: spec/views/posts/index.html.erb_spec.rb [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+    end
+
+    it 'shows the post comments, likes count and comments count' do
+      visit user_posts_path(@first_user, @first_post)
+      sleep(3)
+      expect(page).to have_content('Pagination')
+      click_link @first_post.title
+      expect(page).to have_current_path(user_post_path(@first_user, @first_post))
```
Comment (omar-muhamad) [inline spec/views/posts/index.html.erb_spec.rb]: - [x]  Same here, Please put every one of these tests in a separate test case.

File: spec/views/posts/show.html.erb_spec.rb [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+                                       post: @first_post)
+    end
+
+    it 'shows post title, post author, comments_count, likes_count' do
+      visit user_post_path(@first_user, @first_post)
+      sleep(3)
+      expect(page).to have_content(@first_post.title)
+      expect(page).to have_content(@first_post.author.name)
+      expect(page).to have_content("Comments: #{@first_post.comments_counter}")
+      expect(page).to have_content("Likes: #{@first_post.likes_counter}")
```
Comment (omar-muhamad) [inline spec/views/posts/show.html.erb_spec.rb]: - [x]  Same here, Please put every one of these tests in a separate test case.

File: spec/views/posts/show.html.erb_spec.rb [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+    end
+
+    it 'shows post body, username of each commentor and comments' do
+      visit user_post_path(@first_user, @first_post)
+      sleep(3)
+      expect(page).to have_content(@first_post.text)
+      expect(page).to have_content(@first_comment.author.name)
+      expect(page).to have_content(@second_comment.author.name)
+      expect(page).to have_content(@first_comment.text)
+      expect(page).to have_content(@second_comment.text)
```
Comment (omar-muhamad) [inline spec/views/posts/show.html.erb_spec.rb]: - [x]  Same here, Please put every one of these tests in a separate test case.

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @fibini and @dqarias,

Nice job, you are almost there.



## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- Gitflow is used correctly. ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by cwaku — APPROVED** [review]
Hi @fibini 

Your project is complete! There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
 - [x] Correct GitHub Flow 👍🏾 
 - [x] Good commit history 🥇 
 - [x] Good PR title/description 💯 
 - [x] All linters passed 🥇 
 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. The previous reviewer simply missed some of them, and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**